dtfs-central-api/src/v1/routes/middleware/payload-validation/validate-post-submit-deal-cancellation-payload.test.ts

```diff
@@ -0,0 +1,146 @@
+import { AUDIT_USER_TYPES, AuditDetails, MAX_CHARACTER_COUNT, TfmDealCancellation } from '@ukef/dtfs2-common';
+import { generateTfmAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { aTfmSessionUser } from '../../../../../test-helpers';
+import { validatePostSubmitDealCancellationPayload } from './validate-post-submit-deal-cancellation-payload';
+
+describe('validatePostDealCancellationPayload', () => {
+  const getHttpMocks = () => httpMocks.createMocks();
+
+  const aValidPayload = (): { cancellation: TfmDealCancellation; auditDetails: AuditDetails } => ({
+    cancellation: {
+      reason: 'x'.repeat(MAX_CHARACTER_COUNT),
+      bankRequestDate: new Date().valueOf(),
+      effectiveFrom: new Date().valueOf(),
+    },
+    auditDetails: generateTfmAuditDetails(aTfmSessionUser()._id),
+  });
+
+  const invalidPayloads = [
+    {
+      description: 'the cancellation is undefined',
+      payload: { ...aValidPayload(), cancellation: undefined },
+    },
+    {
+      description: "the 'reason' is not a string",
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          reason: 1234,
+        },
+      },
+    },
+    {
+      description: `the 'reason' is over ${MAX_CHARACTER_COUNT} characters`,
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          reason: 'x'.repeat(MAX_CHARACTER_COUNT + 1),
+        },
+      },
+    },
+    {
+      description: "the 'effectiveFrom' is a string",
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          effectiveFrom: new Date().toString(),
+        },
+      },
+    },
+    {
+      description: "the 'effectiveFrom' is undefined",
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          effectiveFrom: undefined,
+        },
+      },
+    },
+    {
+      description: "the 'bankRequestDate' is a string",
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          bankRequestDate: new Date().toString(),
+        },
+      },
+    },
+    {
+      description: "the 'bankRequestDate' is undefined",
+      payload: {
+        ...aValidPayload(),
+        cancellation: {
+          effectiveFrom: undefined,
+        },
+      },
+    },
+    {
+      description: "'auditDetails' is undefined",
+      payload: {
+        ...aValidPayload(),
+        auditDetails: undefined,
+      },
+    },
+    {
+      description: "'auditDetails' is an empty object",
+      payload: {
+        ...aValidPayload().auditDetails,
+        auditDetails: {},
+      },
+    },
+    {
+      description: "'auditDetails.userType' is undefined",
+      payload: {
+        ...aValidPayload(),
+        auditDetails: {
+          ...aValidPayload().auditDetails,
+          userType: undefined,
+        },
+      },
+    },
+    {
+      description: "'auditDetails.id' is invalid and type is tfm",
```

Comment:
Is there a reason for these descriptions to _not_ have "the" at the beginning like the others?

```ts
description: "the 'reason' is ...
description: "the 'effectiveFrom' is ...
description: "the 'bankRequestDate' is ...
description: "'auditDetails' is ...
description: "'auditDetails' is ...
```

I'm wondering if we should add the "the" or remove from the other assertions?

Reply:
Have removed "the" from the other assertions